files/image_config/reset-factory/reset-factory

```diff
@@ -0,0 +1,192 @@
+#!/bin/bash
+###########################################################################
+# SONIC Factory reset script                                              #
+# /usr/bin/reset-factory                                                  #
+# This script is used to reset the system to factory settings.            #
+# It creates factory default configuration and save it to config_db.json. #
+# Also, it clear logs, tech-support, reboot-cause files, warmboot files,  #
+# docker containers non-default users, users history files and            #
+# home directories.                                                       #
+###########################################################################
+
+# Initialize constants
+CONFIG_DB_JSON=/etc/sonic/config_db.json
+DEFAULT_USERS_FILE=/etc/sonic/default_users.json
+PERMLOG=/var/log/systemlog
+SONIC_VERSION=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v build_version)
+SONIC_OVERLAY_UPPERDIR="/host/image-$SONIC_VERSION/rw/etc/sonic"
+
+SERVICES_STOPPED=0
+trap "error_cleanup" HUP INT QUIT PIPE TERM
+
+# Command usage and help
+usage()
+{
+    cat << EOF
+ Usage:  reset-factory < keep-all-config | only-config | keep-basic >
+
+        Create factory default configuration and save it to
+        to ${CONFIG_DB_JSON}.
+        Clears logs, system files and reboot the system.
+
+        Default          - Reset configurations to factory default. Logs and files will be deleted.
+        keep-all-config  - Preserves all configurations after boot. Logs and files will be deleted.
+        only-config      - Reset configurations to factory default. Logs and files will be preserved.
+        keep-basic       - Preserves basic configurations only after boot. Logs and files will be deleted.
+EOF
+}
+
+run_reboot()
+{
+    reboot
+    # If for any reason we reach this code, then force reboot
+    rc=$?
+    if [ $rc -ne 0 ]; then
+        # Force reboot
+        reboot -f
```

Comment:
Why force is required?

Reply:
We have to reboot after reset-factory so we can load the new configurations.
First, we want to try to reboot nicely. If it fails we must force it as we don't want to leave the system in a bad state. ( we stopped sonic.target, deleted containers..)